### ISSUE TITLE
osu!catch auto not working in certain conditions

### ISSUE THREAD

**Issue by Tenexxt** [ppy/osu#29635]
### Type

Game behaviour

### Bug description

Basically the title.
I noticed on [this](https://osu.ppy.sh/beatmapsets/1729139#osu/3533781) maps highest difficulty (converted to !ctb) that the auto not only is playing poorly, it even dies?
I tested it on other converts and also !ctb maps only, and the same issue occurs **under certain circumstances**, or more specifically, stupidly high Circle Size 
On the map [Frost Walker](https://osu.ppy.sh/beatmapsets/2202823#fruits/4662299), i tested for different CS, and here are the results

9.0 - 9.4 CS = 100% Accuracy, no issues.
9.5 - 10 CS = this is where the issues start, the bot tends to miss from time to time. After inspecting them, in noticed that where normaly in the map the bot could get away with standing still since it would still catch the fruit, and now it cant because the CS is too high and the bot doesn't take that into considiration. The higher the CS, the more the same issue appears.

Basically the bot is lazy asf and decides to not move where it should, because it doesn't care if the CS is to high.
Also, other stats don't impact the bot, and if they do it basically doesn't change enough for me to notice.

### Version

osu! version: Lazer 2024.817.0 (latest)

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16788453/compressed-logs.zip)
